src/utils/WidgetUtils.js

```diff
@@ -396,4 +398,25 @@ export default class WidgetUtils {
 
         return capWhitelist;
     }
+
+    static getWidgetSecurityKey(widgetId, widgetUrl, isUserWidget) {
+        let widgetLocation = ActiveWidgetStore.getRoomId(widgetId);
+
+        if (isUserWidget) {
+            const userWidget = WidgetUtils.getUserWidgetsArray()
+                .find((w) => w.id === widgetId && w.content && w.content.url === widgetUrl);
+
+            if (!userWidget) {
+                throw new Error("No matching user widget to form security key");
+            }
+
+            widgetLocation = userWidget.sender;
```

Comment:
Is the sender of a user widget always yourself? (That's what it seems to be for stickerpicker widget, at least...)

Reply:
yes, it has to be. They are stored in account data and other places enforce this assumption.